### ISSUE TITLE
Unable to set permissions for worlds containing numbers or dashes

### ISSUE THREAD

**Issue by bloodmc** [lucko/LuckPerms#166]
As title states, this is an issue with Forge/Vanilla worlds as they use the default format of "DIM" plus ID so nether would be "DIM-1".

**Comment by lucko** [comment]
Yuuup.

Permissions are just stored as a single string. For example:

`some.permission` only applying on server "survival" gets stored in the file as `factions/some.permission`.

`some.other.permission` only applying on server "survival" and in world "nether" gets stored in the file as `factions-nether/some.other.permission`. 

This is the same for temporary permissions, `some.temp.perm` expiring in a few hours from now is stored as `some.temp.perm$1486221311`.

This is just the way I decided to store permissions when I wrote the plugin, it made perfect sense to do so at the time, as it meant for really simple config files.


However, the limitations are pretty annoying, I know. I guess I'll have to do the same thing as with meta nodes, and escape these characters when they get stored. Orrr, just change the layout, which is something I'm not so keen on.

Idk, I'll see what I can do.
